### ISSUE TITLE
State management + fetch_links + save_changes = RevisionIdChanged Exception

### ISSUE THREAD

**Issue by redigaffi** [BeanieODM/beanie#281]
Model:

```
class ResourceExchangeDocument(Document, ResourceExchange):
    created_time: float | None = 0
    metal_usd_prices: List[Link[MetalUsdPriceDocument]] = []

    class Settings:
        name = "resource_exchange"
        use_revision = True
        use_state_management = True
```

Error:
```
data = await ResourceExchangeDocument.all(fetch_links=True).to_list()
metal0 = data.metal_usd_prices[0]
metal0.price = 500
await token_conversion.save_changes()
``` 

When saving, im always getting RevisionIdChanged, If I change fetch_links=True to fetch_links=False it works correctly

**Comment by roman-right** [comment]
Hey. Thank you for the issue. It looks weird. I'll try to fix it this weekend

**Comment by redigaffi** [comment]
any news on this?

**Comment by roman-right** [comment]
I didn't reproduce this bug. Probably it is fixed in `1.11.9` as I changed the way of swapping the revision id. Please try.

If the issue still exists, please add an example, that I can run to see the problem.
